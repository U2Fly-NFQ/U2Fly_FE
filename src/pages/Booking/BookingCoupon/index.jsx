import { Form } from 'antd'
import React from 'react'
import { ButtonOfPage, InputOFPage } from '../../../components'

export default function BookingCoupon() {
  return (
    <div className="booking-coupon">
      <div className="booking-page__container__item__title">
        <h3>Coupon code</h3>
      </div>
      <div className="booking-coupon__container">
        <Form>
          <Form.Item
            name="username"
            style={{
              display: 'inline-block',
              width: '100%',
              margin: '0px',
            }}
            rules={[{ required: true, message: 'Please input your username!' }]}
          >
            <InputOFPage placeholder="Enter coupon code*" />
          </Form.Item>
          <Form.Item>
            <ButtonOfPage title={'Apply voucher'} />
          </Form.Item>
        </Form>
      </div>
    </div>
  )
}
