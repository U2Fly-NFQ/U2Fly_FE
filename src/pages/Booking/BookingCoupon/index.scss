.booking-coupon__container {
  position: relative;
  .fa-circle-check {
    position: absolute;
    top: 17px;
    /* bottom: -4px; */
    z-index: 10;
    right: 10px;
    color: green;
    font-size: 20px;
  }
}
