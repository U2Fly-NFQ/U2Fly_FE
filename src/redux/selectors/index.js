export * from './auth'
export * from './bookingFlightSelector'
export * from './ticket'
