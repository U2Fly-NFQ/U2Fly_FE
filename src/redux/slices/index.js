export * from './searchSlice'
export * from './filterSlice'
export * from './flightSlice'
