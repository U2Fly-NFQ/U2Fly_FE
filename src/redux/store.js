import { configureStore } from '@reduxjs/toolkit'
import bookingFlightsSlice from './slices/bookingFlightsSlice'

// import loginSlice from './slices/loginSlice'
import flightSlice from './slices/flightSlice'
import airportSlice from './slices/airportSlice'

const store = configureStore({
  reducer: {
    // login: loginSlice.reducer,
    bookingFlight: bookingFlightsSlice.reducer,
    flights: flightSlice.reducer,
    airports: airportSlice.reducer,
  },
})

export default store
