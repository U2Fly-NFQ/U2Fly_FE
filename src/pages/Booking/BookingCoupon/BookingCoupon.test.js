import React, { Suspense } from 'react'
import { I18nextProvider } from 'react-i18next'
import { Provider } from 'react-redux'

import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { store } from '../../../redux/store'
import i18n from '../../../translations'
import BookingCoupon from './'
describe('Sub nav bar test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <Router>
              <Suspense>
                <BookingCoupon />
              </Suspense>
            </Router>
          </I18nextProvider>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  // test('it should finish', () => {
  //   render(
  //     <Router>
  //       <Provider store={store}>
  //         <BookingCoupon />
  //       </Provider>
  //     </Router>
  //   )
  //   const form = screen.queryByTestId('bookingCoupon')
  //   const input = screen.queryByTestId('inputDiscount')

  //   fireEvent.change(input, { target: { value: 2 } })
  //   fireEvent.submit(form)
  //   expect(input.value).toBe('2')
  //   expect(window.scrollY).toBe(0)
  // })
})
