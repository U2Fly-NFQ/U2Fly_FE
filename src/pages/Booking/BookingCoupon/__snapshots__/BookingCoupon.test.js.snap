// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sub nav bar test renders correctly 1`] = `
<div
  className="booking-coupon"
>
  <div
    className="booking-page__container__item__title"
  >
    <h3>
      Coupon code
    </h3>
  </div>
  <div
    className="booking-coupon__container"
  >
    <form
      className="ant-form ant-form-horizontal"
      data-testid="bookingCoupon"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="ant-row ant-form-item form-item-coupon ant-form-item-has-success"
        style={
          Object {
            "display": "inline-block",
            "margin": "0px",
            "width": "100%",
          }
        }
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="form-control"
                data-testid="inputDiscount"
                id="idDiscount"
                name="idDiscount"
                onChange={[Function]}
                placeholder="input your discount number"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={
          Object {
            "marginTop": "20px",
          }
        }
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                className="form-submit"
              >
                <button
                  className="btn btn-primary btn-md"
                  id=""
                  type="submit"
                >
                  Apply voucher
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
